x-pack/plugins/security/server/routes/user_profile/get.ts

```diff
@@ -8,26 +8,40 @@
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from '../';
+import type { AuthenticatedUserProfile, UserData } from '../../../common/';
 import { wrapIntoCustomErrorResponse } from '../../errors';
 import { createLicensedRouteHandler } from '../licensed_route_handler';
 
 export function defineGetUserProfileRoute({
   router,
+  getSession,
   getUserProfileService,
 }: RouteDefinitionParams) {
   router.get(
     {
-      path: '/internal/security/user_profile/{uid}',
+      path: '/internal/security/user_profile',
       options: { tags: ['access:accessUserProfile'] },
       validate: {
-        params: schema.object({ uid: schema.string() }),
+        query: schema.object({ data: schema.maybe(schema.string()) }),
       },
     },
     createLicensedRouteHandler(async (context, request, response) => {
+      const session = await getSession().get(request);
+      if (!session) {
```

Comment:
Should we put the warn into `getSession().get()` then? I'm catering for this case since the return type includes `undefined`

Reply:
> Should we put the warn into getSession().get() then?

We can, but then it will be a bit less useful for debugging purposes as we won't know in what context session has been requested, but wasn't available. If we have warnings here we'll know for sure that something is requesting user profile when it shouldn't. It's not a big deal though, if you think that these warnings won't be useful feel free to ignore this comment.